pkg/ovs/openflow/ofctrl_bridge_test.go

```diff
@@ -160,3 +166,75 @@ func TestOFBridgePacketRcvd(t *testing.T) {
 		}
 	}
 }
+
+func TestOFMeterStats(t *testing.T) {
+	// Reset OVS metrics (Prometheus) and reinitialize them to test.
+	legacyregistry.Reset()
+	metrics.InitializeOVSMetrics()
+
+	b := NewOFBridge("test-br", GetMgmtAddress(ovsconfig.DefaultOVSRunDir, "test-br"))
+	sw := newFakeOFSwitch(b)
+	b.SetOFSwitch(sw)
+
+	mpMeterStatsReply := openflow15.NewMpReply(openflow15.MultipartType_MeterStats)
+	meterStats1 := openflow15.NewMeterStats(1)
+	stats1 := openflow15.NewMeterBandStats()
+	stats1.PacketBandCount = 100
+	stats1.ByteBandCount = 200
+	meterStats1.AddBandStats(*stats1)
+	meterStats2 := openflow15.NewMeterStats(2)
+	stats2 := openflow15.NewMeterBandStats()
+	stats2.PacketBandCount = 0
+	stats2.ByteBandCount = 0
+	meterStats2.AddBandStats(*stats2)
+	// Unexpected case
+	meterStats3 := openflow15.NewMeterStats(3)
+	stats3 := openflow15.NewMeterBandStats()
+	stats3.PacketBandCount = 300
+	stats3.ByteBandCount = 400
+	meterStats3.AddBandStats(*stats3)
+	mpMeterStatsReply.Body = append(mpMeterStatsReply.Body, meterStats1)
+	mpMeterStatsReply.Body = append(mpMeterStatsReply.Body, meterStats2)
+	mpMeterStatsReply.Body = append(mpMeterStatsReply.Body, meterStats3)
+	expectedOVSMeterPacketCount := `
+	# HELP antrea_agent_ovs_meter_packet_dropped_count [ALPHA] Number of packets dropped by OVS meter. The value is more than 0 when the packets exceed the rate-limit.
+	# TYPE antrea_agent_ovs_meter_packet_dropped_count gauge
+   antrea_agent_ovs_meter_packet_dropped_count{meter_id="PacketInMeterNetworkPolicy"} 100
+   antrea_agent_ovs_meter_packet_dropped_count{meter_id="PacketInMeterTraceflow"} 0
+	`
+	handleMeterStatsReply := func(packetCounts map[int]int64) {
+		for k, v := range packetCounts {
+			switch k {
+			case 1:
+				metrics.OVSMeterPacketDroppedCount.WithLabelValues("PacketInMeterNetworkPolicy").Set(float64(v))
+			case 2:
+				metrics.OVSMeterPacketDroppedCount.WithLabelValues("PacketInMeterTraceflow").Set(float64(v))
+			default:
+				klog.InfoS("Received unexpected meterId", "meterId", k)
+			}
+		}
+	}
+	// Timeout to wait for MeterStats reply
+	b.GetMeterStats(handleMeterStatsReply)
+	b.MultipartReply(sw, mpMeterStatsReply)
+	err := wait.PollImmediate(time.Second, 5*time.Second, func() (done bool, err error) {
```

Comment:
have you considered using `assert.Eventually` here instead?

Reply:
I want to cover the case that we do not receive the corresponding reply and time out eventually. So I prefer to check the timeout error here. For the case that we got the reply, I will change to `assert.Eventually` as suggested.

Reply:
if the expected behavior is "metric is not updated", it's better to sleep for a short duration (maybe 1s here) then check the assertion. There is no need for polling. Besides 5s is a long time to sleep in a unit test. We have 100s of unit tests and using long sleeps has an impact on runtime after a while.

Reply:
Thanks for the suggestion. I remove this timeout case since this may be misunderstanding. I use sleep 1s to test the normal case in the latest change.